staging/src/github.com/openshift/openshift-controller-manager/pkg/build/controller/build/build_controller.go

```diff
@@ -360,10 +373,55 @@ func (bc *BuildController) defaults() builddefaults.BuildDefaults {
 	return copy
 }
 
-func (bc *BuildController) setBuildDefaultProxy(spec *configv1.ProxySpec) {
+func (bc *BuildController) setBuildProxy(buildCfg *configv1.Build, globalProxy *configv1.Proxy) {
 	bc.configLock.Lock()
 	defer bc.configLock.Unlock()
-	bc.buildDefaults.DefaultProxy = spec
+
+	if buildCfg == nil && globalProxy == nil {
+		return
```

Comment:
Make sure the `DefaultProxy` is empty val in this case (looks like you use empty values)

Reply:
so you mean that if both of these are nil, but `DefaultProxy` in the build controller is set to something, empty it out, right @adambkaplan ?

Reply:
if so, part of next push